exchanges/binanceus/binanceus_test.go

```diff
@@ -0,0 +1,1494 @@
+	t.Parallel()
+	_, er := bi.GetSubaccountAssets(context.Background(), "")
+	if !errors.Is(er, errNotValidEmailAddress) {
+		t.Errorf("Binanceus GetSubaccountAssets() expected %v, but found %v", er, errNotValidEmailAddress)
+	}
+	_, er = bi.GetSubaccountAssets(context.Background(), "subaccount@thrasher.io")
+	if er != nil && !strings.Contains(er.Error(), "Illegal request.") {
+		t.Fatal("Binanceus GetSubaccountAssets() error", er)
+	}
+}
+
+func TestGetOrderRateLimits(t *testing.T) {
+	t.Parallel()
+	_, er := bi.GetOrderRateLimits(context.Background(), 0)
+	if er != nil {
+		t.Error("Binanceus GetOrderRateLimits() error", er)
+	}
+}
+
+func TestNewOrderTest(t *testing.T) {
+	t.Parallel()
+
+	req := &NewOrderRequest{
+		Symbol:      currency.NewPair(currency.LTC, currency.BTC),
+		Side:        order.Buy.String(),
+		TradeType:   BinanceRequestParamsOrderLimit,
+		Price:       0.0025,
+		Quantity:    100000,
+		TimeInForce: BinanceRequestParamsTimeGTC,
+	}
+	_, err := bi.NewOrderTest(context.Background(), req)
+	switch {
+	case areTestAPIKeysSet() && err != nil:
+		t.Error("Binanceus NewOrderTest() error", err)
+	case !areTestAPIKeysSet() && err == nil && !mockTests:
+		t.Error("Binanceus NewOrderTest() expecting an error when no keys are set")
+	case mockTests && err != nil:
+		t.Error("Binanceus Mock NewOrderTest() error", err)
+		// default:
+		// 	t.Error("Binanceus NewOrderTest() error", err)
+	}
+	req = &NewOrderRequest{
+		Symbol:        currency.NewPair(currency.LTC, currency.BTC),
+		Side:          order.Sell.String(),
+		TradeType:     BinanceRequestParamsOrderMarket,
+		Price:         0.0045,
+		QuoteOrderQty: 10,
+	}
+
+	result, err := bi.NewOrderTest(context.Background(), req)
+	switch {
+	case areTestAPIKeysSet() && err != nil:
+		t.Error("NewOrderTest() error", err)
+	case !areTestAPIKeysSet() && err == nil && !mockTests:
+		t.Error("NewOrderTest() expecting an error when no keys are set")
+	case mockTests && err != nil:
+		t.Error("Mock NewOrderTest() error", err)
+	}
+	re, _ := json.Marshal(result)
+	println(string(re))
+}
+
+func TestGetOrder(t *testing.T) {
+	t.Parallel()
+
+	_, er := bi.GetOrder(context.Background(), OrderRequestParams{})
+	if !errors.Is(er, errIncompleteArguments) {
+		t.Errorf("Binanceus GetOrder() error expecting %v, but found %v", errIncompleteArguments, er)
+	}
+	_, er = bi.GetOrder(context.Background(), OrderRequestParams{
+		Symbol:            "BTCUSDT",
+		OrigClientOrderId: "something",
+	})
+	// You can check the existance of an order using a valid Symbol and OrigClient Order ID
+	if er != nil && !strings.Contains(er.Error(), "Order does not exist.") {
+		t.Error("Binanceus GetOrder() error", er)
+	}
+}
+
+func TestGetAllOpenOrders(t *testing.T) {
+	t.Parallel()
+	orders, er := bi.GetAllOpenOrders(context.Background(), "")
+	if er != nil {
+		t.Error("Binanceus GetAllOpenOrders() error", er)
+	}
+	ordersString, _ := json.Marshal(orders)
+	println(string(ordersString))
```

Comment:
I will stop commenting but please remove code that doesn't not get checked in test.